File: backend/app/api/schemas/inbox.py
from typing import Any, Dict, List, Optional
from pydantic import BaseModel


class InboxItemOut(BaseModel):
    id: int
    user_id: int
    category: str
    title: str
    body: str
    metadata: Optional[Dict[str, Any]] = None
    is_read: bool
    created_at: str


class InboxListOut(BaseModel):
    count: int
    items: List[InboxItemOut]


class InboxMarkReadIn(BaseModel):
    user_id: int
    item_id: int


class InboxMarkAllReadIn(BaseModel):
    user_id: int

File: frontend/src/types/Inbox.ts
export type InboxCategory = "reminder" | "insight" | "experiment" | "safety" | "system";

export interface InboxItem {
  id: number;
  user_id: number;
  category: InboxCategory | string;
  title: string;
  body: string;
  metadata?: Record<string, any> | null;
  is_read: boolean;
  created_at: string; // ISO
}

export interface InboxListResponse {
  count: number;
  items: InboxItem[];
}

